src/backports/concepts.h

```diff
@@ -111,12 +111,30 @@ namespace ql::concepts {
 #ifdef QLEVER_CPP_17
 using namespace ::concepts;
 using ::ranges::bidirectional_iterator;
+using ::ranges::constructible_from;
 using ::ranges::contiguous_iterator;
+using ::ranges::equality_comparable_with;
 using ::ranges::forward_iterator;
+using ::ranges::indirectly_copyable;
 using ::ranges::input_iterator;
+using ::ranges::integral;
+using ::ranges::invocable;
 using ::ranges::random_access_iterator;
+using ::ranges::same_as;
 using ::ranges::sentinel_for;
 using ::ranges::sized_sentinel_for;
+using ::ranges::unsigned_integral;
+using ::ranges::weakly_incrementable;
+
+template <typename T>
+CPP_requires(default_initializable, requires(const T& t)(T{}, ::new T));
+
+template <typename T>
+CPP_concept default_initializable = CPP_requires_ref(default_initializable, T);
+
+template <typename T>
```

Comment:
There is already ad_utility::FloatingPoint, an equivalent to C++20 std::floating_point... it seems this is a good place for this concept.